common/src/app/common/logic/libraries.cljc

```diff
@@ -1624,6 +1624,56 @@
       (seq applied-tokens)
       (update-tokens container dest-shape origin-shape applied-tokens))))
 
+(defn- generate-update-attrs
```

Comment:
I propose the name `add-update-attr-changes`. Although it ends generating some changes objects, I think this is not a typical `generate-xx` function, because this one is lower level (it receives directly the operations it needs to add).